docs/source/DeveloperDocumentation/TaskEngineWorkflows.md

```diff
@@ -839,6 +839,143 @@ Job callbacks are triggered when the entire job has completed. Below is a list o
+| custom_data       |No | This field accepts consumer custom data (such as consumer internal reference ) and returns it as part of the job callback. | |
+
+### Medatailor Channel Assembly: Payload example
+
+```json
+{
+  "client": "demo-client",
+  "job": {
+    "workflow": "mediatailor_channel_assembly"
+  },
+  "parameters": {
+    "content_id": "demo-content",
+    "clips": [
+      {
+        "source": "https://cdn.com/assets/1.m3u8"
+      },
+      {
+        "source": "https://cdn.com/assets/2.m3u8"
+      },
+      {
+        "source": "https://cdn.com/assets/3.m3u8"
+      }
+    ],
+    "rest_endpoints": [
+      "http://your.custom.endpoint"
+    ]
+  }
+}
+```
+
+### Medatailor Channel Assembly: Callback properties
+
+#### Task Callback
+
+Task callbacks are triggered after each task within a workflow is completed. Below is a list of the default properties for the callback:
+
+| Property Name     | Required |
+| ----------------- | -------- |
+| job_id            | Unique job identifier generated by the Task Engine. |
+| task_id           | Unique task identifier generated by the Task Engine. |
+| task_name         | Name of the task that triggered the callback. |
+| workflow          | Name of the workflow being executed. |
+| event             | This will identify the event that caused the callback to be triggered. It can be one of `start`, `complete` or `fail`. |
+| content_id        | Content ID provided when the job was submitted. |
+| message           | Any message associated with the event. This will usually contain exception messages. |
+
+#### Job Callback
+
+Job callbacks are triggered when the entire job has completed. Below is a list of the default properties for the callback.
+
+| Property Name     | Required |
+| ----------------- | -------- |
+| job_id            | Unique job identifier generated by the Task Engine. |
+| status            | This will identify the status of the job. It can be either `completed` or `failed`. |
+| workflow          | Name of the workflow being executed. |
+| content_id        | Content ID provided when the job was submitted. |
+| message           | Resulting live streaming URL |
+| custom_data       | Returns the custom data submitted to the workflow. |
+
+## MEDIATAILOR CHANNEL STATE
+
+This workflow allows for stopping and starting mediatailor channels.
+
+### Medatailor Channel State: Parameters
+
+| Parameter Name    | Required |  Description | Default |
+| ----------------- | -------- | ------------ | ------- |
+| workflow          |Yes| Specify 'mediatailor_channel_assembly'. ||
+| content_id        |Yes| Unique identifier of the content. This is usually a key that allows identification of the content in the client’s system. ||
+| state             |Yes| Either `start` or `stop` ||
+| delete            |No | This boolean indicates whether the channel must be deleted or not |false|
```

Comment:
Perhaps more my curiosity than documentation, but how does this work in tandem with `start`? Might be worth documenting any restrictions, albeit would be pretty silly to set to true with `start` but 🤷

Reply:
It's more there to support stop and delete instead of just stopping or always deleting when stopping a channel. 

If a consumer sets the state to `start` and then opt to set `delete` to false, it either does what the consumer requested or should be handled programatically (ignored or throws an exception).